src/libvirtApi/domain.js

```diff
@@ -1112,3 +1113,23 @@ export function domainUpdateDiskAttributes({ connectionName, objPath, target, re
                 return call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
             });
 }
+
+export async function domainRemoveSpice({ connectionName, id: objPath }) {
+    /* Ideally this would be done by virt-xml, but it doesn't offer that functionality yet
+     * see https://issues.redhat.com/browse/RHEL-17436 */
+    const [domXML] = await call(connectionName, objPath, 'org.libvirt.Domain', 'GetXMLDesc', [Enum.VIR_DOMAIN_XML_INACTIVE], { timeout, type: 'u' });
+    const updatedXML = removeSpice(domXML);
+
+    // check that updatedXML is valid; if not, throw; it needs to be updated manually
+    await (cockpit.spawn(["virt-xml-validate", "-"], { err: "message" }).input(updatedXML));
+
+    await call(connectionName, objPath, 'org.libvirt.Domain', 'Undefine', [0], { timeout, type: 'u' });
+    try {
+        await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [updatedXML], { timeout, type: 's' });
+    } catch (ex) {
+        // if this fails, we are in trouble; but at least try to restore the original XML
+        console.warn("domainRemoveSpice defining updated XML", updatedXML, "failed, restoring original:", ex);
+        await call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'DomainDefineXML', [domXML], { timeout, type: 's' });
+        throw ex;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1387-20240117-024845-4e8cc37f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)